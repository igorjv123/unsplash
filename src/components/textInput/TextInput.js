import React from 'react';

const TextInput = (props) => {
    const { value, onChange, placeholder, className } = props;
    return(
        <input type="text" className={className} defaultValue={value} placeholder={placeholder} onChange={onChange}/>
    )
};

export default TextInput;