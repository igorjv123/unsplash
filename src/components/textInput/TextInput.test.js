import React from 'react';
import TextInput from './TextInput';
import { shallow, render } from 'enzyme';

describe('TextInput', () => {
    const onChangeFn = jest.fn(() => 'change');
    let placeholder = 'placeholder';
    let className = 'class';
    let value = 'val';

    beforeEach(() => {
        placeholder = 'placeholder';
        className = 'class';
        value = 'val';
    })

    it('should call onChange prop function', () => {
        const wrapper = shallow(<TextInput onChange={onChangeFn}/>);
        const input = wrapper.find('input')

        input.simulate('change');

        expect(onChangeFn).toHaveBeenCalled();
    });

    it('should set attribs from props', () => {
        const wrapper = render(<TextInput
            placeholder={placeholder}
            className={className}
            value={value}
        />);

        expect(wrapper).toMatchSnapshot();
    });
});
