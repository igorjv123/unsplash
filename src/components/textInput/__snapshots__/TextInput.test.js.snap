// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should set attribs from props 1`] = `
<input
  class="class"
  placeholder="placeholder"
  type="text"
  value="val"
/>
`;
